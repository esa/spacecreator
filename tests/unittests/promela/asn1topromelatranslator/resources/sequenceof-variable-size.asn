SequencesOf DEFINITIONS ::= BEGIN

Type1 ::= INTEGER(0..3)

SimpleVariableSizeSequenceOf ::= SEQUENCE (SIZE(3..5)) OF Type1

END