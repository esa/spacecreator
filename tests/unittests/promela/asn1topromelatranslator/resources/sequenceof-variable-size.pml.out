inline Type1_generate_value(value)
{
    if
    ::  true;
        value = 0;
    ::  true;
        value = 1;
    ::  true;
        value = 2;
    ::  true;
        value = 3;
    fi;
}
inline SimpleVariableSizeSequenceOf_generate_value(value)
{
    int i;
    if
    ::  true;
        for(i : 0 .. 3)
        {
            Type1_generate_value(value.data[i]);
        }
    ::  true;
        for(i : 0 .. 4)
        {
            Type1_generate_value(value.data[i]);
        }
    ::  true;
        for(i : 0 .. 5)
        {
            Type1_generate_value(value.data[i]);
        }
    fi;
}
